src/util.ts

```diff
@@ -844,3 +844,32 @@ export function makeMemoizedByteOffsetConverter(buffer: Buffer): (byteOffset: nu
 	};
 }
 
+export class TempFileProvider {
+	private static globalState: vscode.Memento;
+
+	/**
+	 * register store to provider for persistance
+	 */
+	static registerStore(globalState: vscode.Memento) {
+		TempFileProvider.globalState = globalState;
+	}
+
+	/**
+	 * returns path to temp file with name
+	 */
+	static getFilePath(name: string): string {
+		let tempDir = TempFileProvider.globalState.get<string>('tempDir');
```

Comment:
One corner case is that `getFilePath` gets called when the globalState is not yet set. This may not happen now, but when the code that registers the store gets moved around in the future, it can happen.

Reply:
I've added a comment to above `TempFileProvider.registerStore must be called be for this`.

One work around could be creating a new temp dir if globalState is undefined (and storing it locally on the object). Thoughts?